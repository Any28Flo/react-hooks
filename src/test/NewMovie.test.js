import React from "react";
import {render, cleanup} from "@testing-library/react";
import NewMovie from "../components/NewMovie";

afterEach(cleanup);

test('<NewMovie/>' , () =>{
   const {debug, getByTestId, queryByTestId, container} = render(<NewMovie/>);

    expect(getByTestId('page-title').textContent).toBe('New Movie');
    /*Integration test*/
    expect(queryByTestId('movie-form')).toBeTruthy();

    /* toMatchSnapshot method creates a new folder __snapshots__
    * Inside you could find json object DOM data of your component
    * This method we could use it to check if something is correcltyl rendered
    * */
    /* We could change the DOM element and to rebuild de json map of our component
    We press u inside our terminal
    * */


    expect(container.firstChild).toMatchSnapshot();

});
