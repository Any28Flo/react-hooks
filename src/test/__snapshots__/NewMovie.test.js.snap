// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewMovie/> 1`] = `
<div>
  <h1
    data-testid="page-title"
  >
    New Movie
  </h1>
  <form
    data-testid="movie-form"
  >
    <label>
      Movie name:
    </label>
    <input
      type="text"
    />
    <button>
      Submit
    </button>
  </form>
</div>
`;
